### PR TITLE
chore: improve errors on sentry

### PR DESCRIPTION
**Pull request by johangirod**
- [x] refactor: use nextJS sentry in front when useReact
  - Add a common entry in front that is used in pages with react (which doesn't contain sentry browser init)
  - Remove frontend index.js load in page with react enabled
- [x] chore: use default nextJS config for sentry
  - use server.config.ts, client.config.ts and edge.config.ts like in the doc
  - add a withSentryConfig to next.config.js, which select the right config file depending on the env and add sourcemap to sentry
  - replace isSentryInitialized by isSentryActivated
  - the front & back error will now be sent to the sentry project : annuaire-entreprises-back (we might want to rename it to annuaire-entreprises-site)
- [x] [fix: log error instead of message in redis-storage](https://github.com/etalab/annuaire-entreprises-site/pull/726/commits/9d5ff39abfcb0e29603579c7b4d5cbc2fa3b0cd9) so that we have the stacktrace and explicit message in sentry
- [ ] ci: add new env variable to GitHub & production machine

You can review commit by commit for this PR.

### CODE REVIEW

File: frontend/src/load-bar.js [hunk 1]
```diff
@@ -40,9 +40,6 @@ const init = () => {
 };
 
 const loadBarFactory = () => {
-  const _loader = null;
-  let _currentJobId = null;
```
Comment (XavierJp) [inline frontend/src/load-bar.js:44]: Why delete these two var ?

File: frontend/src/common.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import './helpers-copy-paste';
+import './helpers-feedbacks';
+import './load-bar';
```
Comment (XavierJp) [inline frontend/src/common.js:3]: Great. 

I believe we also need front state machine and inpi pdf download

Can you add a detailled comment to explain the purpose of this file and distinguish it from `/index` ?

File: frontend/src/sentry.js [hunk 3]
```diff
@@ -1,8 +1,10 @@
 import * as Sentry from '@sentry/browser';
 
-var dsn = import.meta.env.VITE_SENTRY_FRONT_DSN;
+const dsn = import.meta.env.VITE_SENTRY_FRONT_DSN;
+export const isSentryActivated = import.meta.env.PROD && Sentry && dsn
```
Comment (XavierJp) [inline frontend/src/sentry.js:4]: I fear there might be a possible confusion between the two `isSentryActivated` maybe we could have distinct names like `isSentryReactActivated`vs `isSentryViteActivated`

File: sentry.edge.config.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import * as Sentry from '@sentry/nextjs';
+import { isSentryActivated } from '#utils/sentry';
+if (isSentryActivated) {
+  Sentry.init({
+    dsn: process.env.SENTRY_DSN,
+    tracesSampleRate: 0.1,
+    maxBreadcrumbs: 0, // dont log breadcrumb
+  });
+}
```
Comment (XavierJp) [inline sentry.edge.config.ts:9]: Is it necessary when we dont have edge functions ?

File: frontend/src/load-bar.js (hunk 1 above)
Comment (johangirod) [reply frontend/src/load-bar.js:44]: They were listed as not used by typescript, so I removed them without thinking. And indeed, const & let are not assigned to `this`.

File: sentry.edge.config.ts (hunk 4 above)
Comment (johangirod) [reply sentry.edge.config.ts]: I agree, it shouldn't be there. There was a warning without it, but I'm going to investigate further and see if we can do without.

File: frontend/src/common.js (hunk 2 above)
Comment (johangirod) [reply frontend/src/common.js]: https://github.com/etalab/annuaire-entreprises-site/pull/726/commits/f6c8ff2c5057f6f16de9d16004aebd5913743559#diff-ff1d61de8655c32a447df582516565f722b8f17d0054fade55829de6800d9a2fR1-R10

File: sentry.edge.config.ts (hunk 4 above)
Comment (johangirod) [reply sentry.edge.config.ts]: Well, I just removed it, the warning is just a warning that we can safely ignore while we're not using Edge functions

**Comment by XavierJp** [comment]
Thanks I will test it, add env variable and merge it tomorrow

Review app are going to be a blast for this kind of PR

**Review by XavierJp — APPROVED** [review]

**Pull request merged**